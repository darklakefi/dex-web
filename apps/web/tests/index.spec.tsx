import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Page from "../src/app/(swap)/page";

describe("Page", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<Page />);
    expect(baseElement).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
