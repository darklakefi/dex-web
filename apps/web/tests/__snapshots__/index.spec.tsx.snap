// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page > should render successfully 1`] = `
<body>
  <div>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="container"
        >
          <div
            class="flex flex-col items-center justify-center"
          >
            <p
              class="font-normal font-sans text-3xl text-green-100 leading-8.5 tracking-wider animate-bounce"
            >
              Under construction ðŸš§
            </p>
            <div
              class="w-sm"
            >
              <div
                class="flex w-full flex-col items-center gap-10 bg-transparent"
              >
                <div
                  class="mb-2 flex-1 items-stretch justify-start text-green-200 p-5 shadow-green-900 shadow-sm flex w-full flex-col gap-5 bg-green-800"
                >
                  <div
                    class="flex w-full items-center justify-between gap-2"
                  >
                    <div
                      class="flex items-center gap-2"
                    >
                      <svg
                        class="text-green-100 fill-current size-4"
                        fill="none"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M22 7V6H20V7H19V9H20V10H19V11H18V12H16V11H15V9H14V7H13V6H14V4H13V3H11V4H10V6H11V7H10V9H9V11H8V12H6V11H5V10H4V9H5V7H4V6H2V7H1V9H2V10H3V14H4V17H5V19H6V21H18V19H19V17H20V14H21V10H22V9H23V7H22ZM18 14V17H17V19H7V17H6V14H5V13H6V14H8V13H9V12H10V11H11V9H13V11H14V12H15V13H16V14H18V13H19V14H18Z"
                          fill="#1A9A56"
                        />
                      </svg>
                      <p
                        class="font-normal font-sans text-green-200 text-lg leading-6 tracking-wider"
                      >
                        Featured Pools
                      </p>
                    </div>
                    <div
                      class="flex items-center gap-2"
                    >
                      <p
                        class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                      >
                        24H APR
                      </p>
                      <svg
                        class="size-6 fill-current text-green-300"
                        id="chevron-down"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polygon
                          points="5 7 7 7 7 8 8 8 8 9 9 9 9 10 10 10 10 11 11 11 11 12 13 12 13 11 14 11 14 10 15 10 15 9 16 9 16 8 17 8 17 7 19 7 19 8 20 8 20 10 19 10 19 11 18 11 18 12 17 12 17 13 16 13 16 14 15 14 15 15 14 15 14 16 13 16 13 17 11 17 11 16 10 16 10 15 9 15 9 14 8 14 8 13 7 13 7 12 6 12 6 11 5 11 5 10 4 10 4 8 5 8 5 7"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="flex w-full flex-col gap-3"
                  >
                    <div
                      class="flex w-full justify-between bg-green-700 p-3"
                    >
                      <div
                        class="flex text-green-300"
                      >
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          SOL
                        </p>
                        /
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          USDC
                        </p>
                      </div>
                      <p
                        class="font-normal font-sans text-lg leading-6 tracking-wider text-green-100"
                      >
                        100
                        %
                      </p>
                    </div>
                    <div
                      class="flex w-full justify-between bg-green-700 p-3"
                    >
                      <div
                        class="flex text-green-300"
                      >
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          SOL
                        </p>
                        /
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          USDC
                        </p>
                      </div>
                      <p
                        class="font-normal font-sans text-lg leading-6 tracking-wider text-green-100"
                      >
                        120.12
                        %
                      </p>
                    </div>
                    <div
                      class="flex w-full justify-between bg-green-700 p-3"
                    >
                      <div
                        class="flex text-green-300"
                      >
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          ETH
                        </p>
                        /
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          USDT
                        </p>
                      </div>
                      <p
                        class="font-normal font-sans text-lg leading-6 tracking-wider text-green-100"
                      >
                        85.5
                        %
                      </p>
                    </div>
                    <div
                      class="flex w-full justify-between bg-green-700 p-3"
                    >
                      <div
                        class="flex text-green-300"
                      >
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          BTC
                        </p>
                        /
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          USDC
                        </p>
                      </div>
                      <p
                        class="font-normal font-sans text-lg leading-6 tracking-wider text-green-100"
                      >
                        150.75
                        %
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="mb-2 flex-1 items-stretch justify-start text-green-200 p-5 shadow-green-900 shadow-sm flex w-full flex-col gap-5 bg-green-800"
                >
                  <div
                    class="flex w-full items-center justify-between gap-2"
                  >
                    <div
                      class="flex items-center gap-2"
                    >
                      <svg
                        class="text-green-100 fill-current size-4"
                        id="fire-solid"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m19,13v-3h-1v-1h-1v-3h-1v-2h-1v-1h-1v-1h-1v-1h-2v1h1v2h-1v2h-1v1h-1v1h-1v1h-1v1h-1v3h1v2h-1v-1h-1v-2h-1v2h-1v3h1v2h1v1h1v1h1v1h1v1h8v-1h1v-1h1v-1h1v-2h1v-5h-1Zm-2,7h-1v1h-2v1h-4v-1h-1v-4h1v-1h1v-1h1v-1h1v-3h-1v-1h-1v-1h1v1h2v2h1v5h-1v2h1v-1h1v-1h1v3Z"
                        />
                      </svg>
                      <p
                        class="font-normal font-sans text-green-200 text-lg leading-6 tracking-wider"
                      >
                        Trending Pools
                      </p>
                    </div>
                    <div
                      class="flex items-center gap-2"
                    >
                      <p
                        class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                      >
                        24H APR
                      </p>
                      <svg
                        class="size-6 fill-current text-green-300"
                        id="chevron-down"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polygon
                          points="5 7 7 7 7 8 8 8 8 9 9 9 9 10 10 10 10 11 11 11 11 12 13 12 13 11 14 11 14 10 15 10 15 9 16 9 16 8 17 8 17 7 19 7 19 8 20 8 20 10 19 10 19 11 18 11 18 12 17 12 17 13 16 13 16 14 15 14 15 15 14 15 14 16 13 16 13 17 11 17 11 16 10 16 10 15 9 15 9 14 8 14 8 13 7 13 7 12 6 12 6 11 5 11 5 10 4 10 4 8 5 8 5 7"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="flex w-full flex-col gap-3"
                  >
                    <div
                      class="flex w-full justify-between bg-green-700 p-3"
                    >
                      <div
                        class="flex text-green-300"
                      >
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          SOL
                        </p>
                        /
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          USDC
                        </p>
                      </div>
                      <p
                        class="font-normal font-sans text-lg leading-6 tracking-wider text-green-100"
                      >
                        100
                        %
                      </p>
                    </div>
                    <div
                      class="flex w-full justify-between bg-green-700 p-3"
                    >
                      <div
                        class="flex text-green-300"
                      >
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          SOL
                        </p>
                        /
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          USDC
                        </p>
                      </div>
                      <p
                        class="font-normal font-sans text-lg leading-6 tracking-wider text-green-100"
                      >
                        120.12
                        %
                      </p>
                    </div>
                    <div
                      class="flex w-full justify-between bg-green-700 p-3"
                    >
                      <div
                        class="flex text-green-300"
                      >
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          ETH
                        </p>
                        /
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          USDT
                        </p>
                      </div>
                      <p
                        class="font-normal font-sans text-lg leading-6 tracking-wider text-green-100"
                      >
                        85.5
                        %
                      </p>
                    </div>
                    <div
                      class="flex w-full justify-between bg-green-700 p-3"
                    >
                      <div
                        class="flex text-green-300"
                      >
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          BTC
                        </p>
                        /
                        <p
                          class="font-normal font-sans text-lg leading-6 tracking-wider text-green-300"
                        >
                          USDC
                        </p>
                      </div>
                      <p
                        class="font-normal font-sans text-lg leading-6 tracking-wider text-green-100"
                      >
                        150.75
                        %
                      </p>
                    </div>
                  </div>
                </div>
                <button
                  class="inline-flex items-center justify-center gap-2 font-medium uppercase leading-none outline-offset-1 focus:outline-2 focus:outline-solid transition-colors duration-200 disabled:cursor-not-allowed disabled:opacity-50 [&>*]:flex [&>*]:items-center [&>*]:leading-none px-3 py-1.5 text-green-300 hover:text-green-200 focus:text-green-200 focus:outline-white w-fit"
                >
                  <div
                    class="font-normal font-sans text-lg leading-6 tracking-wide underline text-inherit"
                  >
                    explore all pools
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
